packages/react-components/src/components/chart/hooks/useDataZoomListeners.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType, getInstanceByDom } from 'echarts';
+
+export const useDataZoomListeners = (ref: React.RefObject<HTMLDivElement>) => {
+  const { setViewport } = useViewport();
+
+  useEffect(() => {
+    if (ref.current) {
+      const chart: EChartsType | undefined = getInstanceByDom(ref.current);
```

Comment:
is it better to get the ref from the dom? I think we also have the chartRef available from the useEChart hook